### ISSUE TITLE
Allow N in #[ink(selector = N)] to be a numeric

### ISSUE THREAD

**Issue by Robbepop** [paritytech/ink#681]
Currently we only support `N` to be a string literal in the form of a hex encoded `u32` such as `"0xC0DECAFE"`.
This issue tracks progress in adding the possibility for `N` to be a raw `u32` literal.
The value of the raw `u32` literal is then converted to 4 bytes using `to_be_bytes`.

Why big endian instead of little endian which is the norm for Wasm based execution?

Consider the following code snippets:

```rust
println!("A = {:?}", 0xC0DECAFE_u32.to_le_bytes());
println!("B = {:?}", 0xC0DECAFE_u32.to_be_bytes());
println!("C = {:?}", [0xC0_u8, 0xDE, 0xCA, 0xFE]);
```
While `B` and `C` are printing the same, `A` has reversed order which is a potential confusion for users.
Therefore using big endian encoding is the way to go. We should make sure to document this properly.

Implementing raw `u32` for `selector` might even result in us deprecating the previous hex-encoded string since it superseeds its functionality and is also slightly simpler to use: no `"` bracketing.

Before:

```rust
#[ink(message, selector = "0xC0DECAFE")]
fn my_message(&self);
```

After:

```rust
#[ink(message, selector = 0xC0DECAFE)]
fn my_message(&self);
```
or
```rust
#[ink(message, selector = 42)]
fn my_message(&self);
```

**Comment by xgreenx** [comment]
Hmm, seems it is implemented?

**Comment by HCastano** [comment]
Yep, you're right.

Closed by #928.
